crates/bevy_animation/src/lib.rs

```diff
@@ -693,8 +725,12 @@ fn apply_animation(
                 let lerp = (animation.seek_time - ts_start) / (ts_end - ts_start);
 
                 // Apply the keyframe
-                match &curve.keyframes {
-                    Keyframes::Rotation(keyframes) => {
+                match (&curve.interpolation, &curve.keyframes) {
```

Comment:
This function is already huge, can we factor out these changes into smaller functions (without a major impact to readability or performance)?